scripts/c8y-publish-release.sh

```diff
@@ -11,13 +11,15 @@ if [ $# -eq 0 ]; then
     echo "missing required positional argument: TAG" >&2
     echo
     echo "Usage:"
-    echo
-    echo "    $0 <TAG>"
-    echo
+    echo 
+    echo "    $0 <TAG> <PRERELEASE>=true/false "
+    echo 
+    echo "    <PRERELEASE> is optional to update a draft to a prerelease "
     exit 1
 fi
 
 TAG="$1"
+PRERELEASE="$2"
```

Comment:
Since this is an "optional" value it would make more sense to have a flag for this, e.g. `--pre-release`. Otherwise the user might be forced to provide the value if the bash options `-u` is provided.

Reply:
Here is an example of one of the common ways to parse arguments in bash:

```sh
#!/usr/bin/env bash
set -e

help() {
    cat << EOT
Publish github releases

USAGE
    $0 <TAG> [OPTIONS]

FLAGS
    --pre-release       Publish release as a pre-release (only if it set to draft)

EXAMPLES
    $0 1.0.0 --pre-release

EOT
}

# Defaults
PRE_RELEASE=0

# Parse arguments
REST_ARGS=()
while [ $# -gt 0 ]; do
    case "$1" in
        -h|--help)
            help
            exit 0
            ;;

        --pre-release)
            PRE_RELEASE=1
            ;;

        *)
            REST_ARGS+=("$1")
            ;;
    esac
    shift
done

# Only set if rest arguments are defined
if [ ${#REST_ARGS[@]} -gt 0 ]; then
    set -- "${REST_ARGS[@]}"
fi

TAG=
if [ $# -eq 0 ]; then
    echo "Missing required argument" >&2
    help
    exit 1
fi
TAG="$1"
```

Reply:
Just testing with this:
if [[ ($* == -p) || ($* == --pre-release) ]]; then

But can also use your approach later

Reply:
using the while and case statement is easy to use an extend/maintain etc..

Reply:
and the above example does not just do parsing, but also help etc.

Reply:
Used it in my last commit